ee/agent/internal/informer_stacks.go

```diff
@@ -24,74 +24,97 @@ func GetExpectedModules(stackName string, stacks InMemoryStacksModules) []string
 }
 
 func NewStackEventHandler(logger sharedlogging.Logger, membershipClient MembershipClient, stacks InMemoryStacksModules) *StackEventHandler {
-	sendStatus := func(stack string, status generated.StackStatus) {
+	sendStatus := func(interpretedStatus generated.StackStatus, stack *unstructured.Unstructured) {
 		if err := membershipClient.Send(&generated.Message{
 			Message: &generated.Message_StatusChanged{
 				StatusChanged: &generated.StatusChanged{
-					ClusterName: stack,
-					Status:      status,
+					ClusterName: stack.GetName(),
+					Status:      interpretedStatus,
+					Statuses:    getStatus(logger, stack),
 				},
 			},
 		}); err != nil {
 			logger.Errorf("Unable to send stack status to server: %s", err)
 		}
 	}
 
-	sendStatusFromStack := func(stack *v1beta1.Stack) {
-		sendStatus(stack.Name, func() generated.StackStatus {
-			if stack.Spec.Disabled {
-				return generated.StackStatus_Disabled
-			}
+	InterpretedStatus := func(stack *unstructured.Unstructured) generated.StackStatus {
+		disabled, found, err := unstructured.NestedBool(stack.Object, "spec", "disabled")
+		if !found || err != nil {
+			panic(err)
+		}
+		if disabled {
+			return generated.StackStatus_Disabled
+		}
 
-			if stack.Status.Ready {
-				for _, module := range GetExpectedModules(stack.Name, stacks) {
-					if !slices.Contains(stack.Status.Modules, module) {
-						return generated.StackStatus_Progressing
-					}
+		ready, found, err := unstructured.NestedBool(stack.Object, "status", "ready")
+		if !found || err != nil {
```

Comment:
You miss `!ready` ?

Reply:
No it's 8 lines, below, it will be deleted in 2 PR

Reply:
Ok, i didn't see the else. Seems useless to add a else later instead of adding to the existing condition.